templates/CRM/Financeextras/Form/Payment/Refund.tpl

```diff
@@ -0,0 +1,66 @@
+<div class="crm-block crm-form-block crm-payment-refund-form-block">
+<div class="crm-accordion-body">
+<div class="message help" >{ts}Refunds take 5-10 days to appear on a customer's statement{/ts}</div></div>
+ <table class="form-layout-compressed">
+        <tr class="crm-payment-refund-form-block-contact_id">
+          <td class="label">{$form.contact.label}</td>
+          <td>{$form.contact.html}</td>
+        </tr>
+        <tr class="crm-payment-refund-form-block-contact_id">
+          <td class="label"><label>{ts}Select Payment To Refund{/ts}</label></td>
+          <td><table class="selector row-highlight payment-info">
+          <tr>
+          <th></th>
+            <th>{ts}Date{/ts}</th>
+            <th>{ts}Original Amount{/ts}</th>
+            <th>{ts}Available Amount{/ts}</th>
+            <th>{ts}Payment Processor{/ts}</th>
+            <th>{ts}Processor Payment ID{/ts}</th>
+          </tr>
+        {foreach from=$paymentInfos item=paymentInfo}
+              <tr  id="tr_{$paymentInfo.financialTrxnId}">
+                <td><input class=" required crm-form-radio" value="{$paymentInfo.transactionId}" type="radio" id="{$paymentInfo.financialTrxnId}" name="payment_row" ></td>
+                <td>{$paymentInfo.date}</td>
+                <td>{$paymentInfo.amount|crmMoney:$paymentInfo.currency}</td>
+                <td class="available_amount_{$paymentInfo.financialTrxnId}">{$paymentInfo.available_amount|crmMoney:$paymentInfo.currency}</td>
+                <td>{$paymentInfo.paymentProcessor}</td>
+                <td>{$paymentInfo.transactionId}</td>
+              </tr>
+          {/foreach}
+</table>
+</td>
+        </tr>
+        <tr  class="crm-payment-refund-form-block-total_amount">
+            <td class="label">{$form.amount.label}</td>
+            <td {$valueStyle}>
+              <span id='totalAmount'>{$form.currency.html|crmAddClass:eight}&nbsp;{$form.amount.html|crmAddClass:eight}</span>
+              {if $freezeFinancialType}
```

Comment:
I am confused with this logic, as I don't see they are being assigned and to see that the UI is incorrect (even the mock). You should not allow selecting a currency there as you cannot refund to different currency.